select.js

```diff
@@ -1,14 +1,39 @@
-define(function() {
-    var init = function() {
-        $('body').on('change', '.c-select', function() {
-            var $container = $(this);
-            var value = $container.find('option:selected').text();
-
-            $container.find('.c-select__value').text(value);
-        });
-    };
+define(['$'], function($) {
+    //
+    // Component constructor
+    //
+    var Select = function Select($el, options) {
+        // Component instance properties. The `$` hash (by convention) caches
+        // the root node and any sub-components.
+        this.value = '';
+        this.dom = {
+            $root: $el,
+            $value: $el.find('.c-select__value')
+        };
+
+        // Event handlers should use $.proxy to preserve access to the instance
+        // as `this` within the handler.
+        this.dom.$root.on('change', $.proxy(this.update, this));
+
+        // Immediately invoke any methods needed to set up initial state.
+        this.update();
+    }
+
+    //
+    // Update the component’s visible value when the underlying select element’s
+    // value changes.
+    //
+    Select.prototype.update = function update() {
+        this.value = this.dom.$root.find('option:selected').text();
+        this.dom.$value.text(this.value);
+        this.dom.$root.trigger('select:update'); // notify subscribers
```

Comment:
Thanks for the rationale. The first reason strikes me as maybe pragmatic, but I also don’t know that jQuery plugins should be our reference point for best practices, especially in 2015 (caveat: I am no expert).

The second reason makes sense, although incidentally that may be why the callbacks bug me: it feels like putting unrelated stuff together (my code is “delegated” to the plugin, where the plugin should just expose an interface). Not sure if that makes sense.